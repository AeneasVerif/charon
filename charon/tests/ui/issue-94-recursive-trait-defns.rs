pub trait Trait1: Sized {
    type T: Trait2;
}

pub trait Trait2: Trait1 {}

pub trait T1<T: T2<Self>>: Sized {}
pub trait T2<T: T1<Self>>: Sized {}

pub trait T3 {
    type T: T5;
}

pub trait T4: T3 {}

pub trait T5 {
    type T: T4;
}

pub trait T6<T: T7>: Sized {
    fn f(x: u64) -> () {
        T::g(x)
    }
}

pub trait T7: T6<Self> {
    fn g(_x: u64) -> ();
}
