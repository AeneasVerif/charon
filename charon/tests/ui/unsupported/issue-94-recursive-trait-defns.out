thread 'rustc' panicked at 'Invalid trait decl group:
test_crate::Trait1
test_crate::Trait2', src/reorder_decls.rs:115:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
